### PR TITLE
Properly address the change default name mapping strategy for @RestHeader

### LINKED ISSUE #13665: RESTEasy Reactive: change default name mapping strategy for `@RestHeader`
ATM if you define your param as `@RestHeader String contentType` it will map to the `contentType` header, but it should map to `Content-Type`, so we have to adjust the name mapping rules to make it work for all HTTP headers by default.

### PR DESCRIPTION
**Pull request by geoand**
Closes: https://github.com/quarkusio/quarkus/issues/13665

### CODE REVIEW

File: extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/headers/HeaderTest.java [hunk 1]
```diff
@@ -27,22 +30,22 @@ public class HeaderTest {
...
 
     @Path("/")
     @ApplicationScoped
     public static class Resource {
         @GET
         public String returnHeaders(@HeaderParam("foo") String header, @HeaderParam("foo2") String header2,
-                @HeaderParam("foo3") String header3, @HeaderParam("foo4") String header4) {
-            return header + ":" + header2 + ":" + header3 + ":" + header4;
+                @HeaderParam("foo3") String header3, @HeaderParam("foo4") String header4, @Context HttpHeaders headers) {
+            return header + ":" + header2 + ":" + header3 + ":" + header4 + ":" + headers.getHeaderString("X-My-Header");
```
Comment (FroMage) [inline extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/headers/HeaderTest.java:48]: I am not entirely sure that this `getHeaderString` is not case-insensitive. In which case it's not testing anything ;)
In which case you'd be better off testing the keys of `getRequestHeaders`

**Review by FroMage — CHANGES_REQUESTED** [review]

File: extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/headers/HeaderTest.java (hunk 1 above)
Comment (geoand) [reply extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/headers/HeaderTest.java:48]: Yeah, you're probably right. That's why I checked with Wireshark, but lemme see if I can fix the test

Comment (geoand) [reply extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/headers/HeaderTest.java]: Should be good to go now

**Review by FroMage — APPROVED** [review]

**Comment by geoand** [comment]
We can do it the other way around, I don't mind at all

**Comment by geoand** [comment]
Given that this passes CI while https://github.com/quarkusio/quarkus/pull/28856 is complaining, I'll go ahead and merge this

**Pull request merged**